### PR TITLE
Use santized-landfill-sample/v3

### PR DESCRIPTION
**Pull request by acmiyaguchi**
`sanitized-landfill-sample/v3` is the most recent version of the sampling script that is now being run daily at 1:00 UTC. There's a bug where running the job more than once doesn't cause the json documents to be overwritten, but the synchronization will only store the last copied file.

This also adds a `sync` command to the integration script. This command translates all of the options into variables that control the `sync.sh` script. This starts to solve issue #19 by adding an interface, but the command itself hasn't been ported. 

I tested this by doing the following:

```bash
$ pipenv shell
$ ./integration.py sync --output-path test-cmd
$ ./integration.py report --data-path test-cmd/data
```

**Changelog**

**1.2.0**
* use sanitized-landfill-sample/v3
* Issue #19: add sync command to integration
* Issue #20: update to python-rapidjson 0.6.1 to include upstream bug-fixes
* reduce noise in report output

**1.2.1**
* Issue #11: typo in TelemetryURISpec
* Issue #14: add documentation for git submodule
* Issue #22: coerce json without content header

### CODE REVIEW

File: integration.py [hunk 1]
```diff
@@ -65,6 +63,7 @@ def post(route, data, content_type):
 
             client = Client()
         else:
+            import app
```
Comment (acmiyaguchi) [inline integration.py]: If the resource folder doesn't exist, then the import will fail at the top-level. There was a chicken-or-egg problem going on here, so I just moved it to the appropriate place in the Reporter class.

File: integration.py [hunk 2]
```diff
@@ -78,11 +77,8 @@ def get_text(self, resp):
         else:
             return resp.data.decode('utf-8')
 
-    def validate_sample(self, namespace, name, messages):
+    def validate_sample(self, namespace, doc_type, doc_version, messages):
         start = time.time()
-        submission, doc_type, doc_version = (
```
Comment (acmiyaguchi) [inline integration.py]: This is the wrong place to be extracting information from the filename, I moved this out into the main portion of the reporter routine.

File: sync.sh [hunk 3]
```diff
@@ -35,7 +35,15 @@ fi
 
 function sync_data {
     src_data_path="s3://${SRC_DATA_BUCKET}/${SRC_DATA_PREFIX}"
-    
+
+    # Use only the most recent data
```
Comment (acmiyaguchi) [inline sync.sh]: Pulling down multiple days worth of data difficult, the interface may look something like this in the future:

```
integration sync all
integration sync head [-n <INT>]
integration sync date <ds_nodash>
```
Until then, synchronization and reporting will only be able to use the currently visible data (which may not necessarily be a bad thing).

File: sync.sh [hunk 4]
```diff
@@ -64,8 +72,8 @@ function sync_data {
         doc_version=$(echo "${path}" | cut -d'/' -f7 | cut -d'=' -f2)
 
         
-        namespace_dir="${data_path}/${namespace}"
-        filename="${submission_date}.${doc_type}.${doc_version}.batch.json"
+        namespace_dir="${data_path}/${submission_date}/${namespace}"
+        filename="${doc_type}.${doc_version}.batch.json"
```
Comment (acmiyaguchi) [inline sync.sh]: I'm changing the file-structure here, I'll probably tag this as a new release afterwards.

**Comment by acmiyaguchi** [comment]
These changes should be appropriate for use in CI given the right permissions.

File: README.md [hunk 5]
```diff
@@ -115,9 +115,12 @@ POST /submit/<namespace>/<docid>/<appName>/<appVersion>/<appUpdateChannel>/<appB
 #### Building from source
 ```bash
 # clone and set the working directory
-$ git clone https://github.com/acmiyaguchi/edge-validator.git
+$ git clone --recursive https://github.com/acmiyaguchi/edge-validator.git
```
Comment (mreid-moz) [inline README.md]: nit: `https://github.com/mozilla-services/edge-validator.git`

File: app.py [hunk 6]
```diff
@@ -98,7 +98,8 @@ def submit(namespace, doctype, docversion=None, **kwargs):
     try:
         docversion = docversion or SCHEMA_VERSIONS[namespace][doctype]
         key = "{}.{}".format(doctype, docversion)
-        NAMESPACE_SCHEMAS[namespace][key](request.data)
+        json_data = request.get_json(force=True)
```
Comment (mreid-moz) [inline app.py]: Why parse, then re-serialize the request data here?

File: integration.py [hunk 7]
```diff
@@ -211,26 +220,80 @@ def _transform(path):
...
+              default='resources/',
+              help="path to the application resource folder.")
+@click.option('--include-data/--ignore-data',
+              default=True,
+              help="fetch sampled data from a remote, performed by default")
+@click.option('--data-bucket', type=str,
+              default='net-mozaws-prod-us-west-2-pipeline-analysis',
+              help="location of the s3 bucket")
+@click.option('--data-prefix', type=str,
+              default='amiyaguchi/sanitized-landfill-sample/v2',
```
Comment (mreid-moz) [inline integration.py]: Should this be updated to `v3` to match the change to `sync.sh` below?

File: app.py (hunk 6 above)
Comment (acmiyaguchi) [reply app.py]: I think when the header isn't set, then the request shows up as form data. This call is consistent whether the content-type header is set or not.

The rapidjson validator accepts strings, so it has to be re-serialized.

Comment (mreid-moz) [reply app.py]: Ok, makes sense - can you add a comment to that effect in the code?

**Comment by acmiyaguchi** [comment]
@mreid-moz thanks!

**Pull request merged**